lib/filequeue.rb

```diff
@@ -52,25 +53,21 @@ def clear
 
   def safe_open(mode)
     File.open(@file_name, mode) do |file|
-      lock file
+   
+     lock file do
       yield file
+     end
+    
     end
   end
 
   # Locks the queue file for exclusive access.
   #
-  # Raises `FileLockError` if unable to acquire a lock.
-  #
-  # Return is undefined.
+  # Raises `Filelock::WaitTimeout` if unable to acquire a lock.
```

Comment:
Can we raise the same exception as before so this isn't a breaking change?

Reply:
That _would_ work, but `Filelock`'s exceptions are somewhat more detailed: 
* [`Filelock::WaitTimeout`](
https://github.com/sheerun/filelock/blob/master/lib/filelock/wait_timeout.rb) - If unable to acquire the lock within the set time (`:wait` option).
* [`Filelock::ExecTimeout`](https://github.com/sheerun/filelock/blob/master/lib/filelock/exec_timeout.rb) - If the `yield`ed block doesn't exit fast enough (`:timeout` option).

`FileLockError` is probably sufficient for the time being, but we may want to move away from it in the future.
What do you think?

Reply:
In general I'd prefer wrapping any FileLock exceptions. Doing so allows us to swap the locking library without breaking our interface contacts. Eg, it allows us to switch from `#flock` to `FileLock`, and maybe at some point in the future to `LockFile`,  without requiring any code changes in clients.

Specifically, `Filelock::ExecTimeout` should never happen based on what we are doing. If it does that would be a bug so we don't need to document that potential exception.